tests/test_target.py

```diff
@@ -0,0 +1,122 @@
+"""
+Unit-tests for target generation
+"""
+
+import click
+import pytest
+import commodore.cluster as cluster
+
+
+cluster_obj = {
+    'id': 'mycluster',
+    'tenant': 'mytenant',
+    'facts': {
+        'distribution': 'rancher',
+        'cloud': 'cloudscale',
+    }
+}
+components = ['test-component']
+catalog = 'ssh://git@git.example.com/cluster-catalogs/mycluster'
+
+
+def test_render_target():
+    target = cluster._full_target(cluster_obj, components, catalog)
+    facts = cluster_obj['facts']
+    assert target != ""
+    all_classes = ([f"defaults.{cn}" for cn in components] +
+                   ['global.common',
+                       f"global.{facts['distribution']}",
+                       f"global.{facts['cloud']}",
+                       f"{cluster_obj['tenant']}.{cluster_obj['id']}",
+                    ])
```

Comment:
Don't use braces here. This can backfire when some developer adds a comma in future. Also, the indentation seems to be out of sync in line 31.